CRM/Utils/File.php

```diff
@@ -1100,4 +1100,45 @@ public static function getExtensionFromPath($path) {
     return pathinfo($path, PATHINFO_EXTENSION);
   }
 
+  /**
+   * Check that filename is a directory.
+   * @param string $fileName
+   * @return bool
+   */
+  public static function isDir($fileName = '') {
+    // If not within open_basedir do not consider it a directory.
+    if (!CRM_Utils_File::isFileInOpenbasedir($fileName)) {
+      return FALSE;
+    }
+    return is_dir($fileName);
+  }
+
+  /**
+   * Check that path is within open_basedir restriction.
+   * @param string $filePath the absolute path
+   * @return bool
+   */
+  private static function isFileInOpenbasedir($filePath = '') {
+    // Use the appropriate comparison function.
+    $compareFunc = 'strncmp';
+    if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
+      $compareFunc = 'strncasecmp';
+    }
+    // Ensure file path is within open_basedir restrictions.
+    $openBasedir = ini_get('open_basedir');
+    if (!empty($openBasedir)) {
+      $baseDirs = explode(PATH_SEPARATOR, $openBasedir);
+      $withinBasedir = FALSE;
+      foreach ($baseDirs as $baseDir) {
+        if ($compareFunc($filepath, $baseDir, strlen($baseDir)) === 0) {
```

Comment:
This should be filePath with a capital P.

I ran the tests against this and this function would need to deal with paths that have `..` or `.` in them, because this algorithm doesn't quite work with them. Using realpath() might work. There is an existing function CRM_Utils_File::isChildPath() but it doesn't work on windows (it only seems to be used lightly for the extension manager).